### PR TITLE
feat(runtime): add Deno.isProxy

### PR DESCRIPTION
**Pull request by kt3k**
This PR exposes `Deno.core.isProxy` as `Deno.isProxy`. This is necessary for implementing `util.types.isProxy` function in `std/node`.

related:
- https://github.com/denoland/deno/issues/15427#issuecomment-1329945670
- https://github.com/denoland/deno/issues/16710

### CODE REVIEW

File: cli/tsc/dts/lib.deno.ns.d.ts [hunk 1]
```diff
@@ -5214,4 +5214,11 @@ declare namespace Deno {
    * @category Runtime Environment
    */
   export function gid(): number | null;
+
+  /**
+   * Returns true if the given value is a proxy.
```
Comment (lucacasonato) [inline cli/tsc/dts/lib.deno.ns.d.ts:5219]:
```suggestion
   * Returns true if the given value is a `Proxy`.
```

**Review by lucacasonato — APPROVED** [review]
Implementation LGTM. As for background: it is highly unlikely TC39 would ever stabilize this as a builtin method.

It is very unfortunate we are adding non I/O methods to the `Deno` global though.

Do note this is a direct public API stabilization - usually we'd have this in unstable for one version. There are no other alternative APIs for this though, so I am fine with immediately stabilizing.

**Review by littledivy — COMMENTED** [review]
LGTM

An alternate suggestion: Move `std/node/util` module to the runtime. It seems like a standalone module so might be easier. We already do this with `node:module` specifier https://github.com/denoland/deno/blob/3f71c0dcf542ccfe1abbd8df82e4cdd1848423cb/cli/module_loader.rs#L67

**Comment by kt3k** [comment]
This is now not necessary because of migration of std/node to ext/node

**Pull request closed without merging**